mpitests/test_apply.py

```diff
@@ -468,3 +543,95 @@ def circuit_mpi():
 
         local_state_vector = circuit_mpi()
         assert np.allclose(local_state_vector, local_expected_output_cpu, atol=tol, rtol=0)
+
+
+class TestExpval:
+    """Tests that expectation values are properly calculated or that the proper errors are raised."""
+
+    @pytest.mark.parametrize(
+        "operation",
+        [
+            qml.PauliX,
+            qml.PauliY,
+            qml.PauliZ,
+            qml.Hadamard,
+            qml.Identity,
+        ],
+    )
+    @pytest.mark.parametrize("wires", [0, 1, 2, numQubits - 3, numQubits - 2, numQubits - 1])
+    def test_expval_single_wire_no_parameters(self, tol, operation, wires):
+        """Tests that expectation values are properly calculated for single-wire observables without parameters."""
+
+        obs = operation(wires)
+        expval_single_wire_no_param(tol, obs)
+
+
+class TestGenerateSample:
```

Comment:
Can the sampling functions also be tested within a QNode? We expect the QNode to be the user facing API.

Reply:
Sure!